harness/tests/storage/test_s3.py

```diff
@@ -41,12 +41,16 @@ def live_manager(tmp_path: Path, require_secrets: bool) -> storage.S3StorageMana
 
 
 @pytest.mark.cloud  # type: ignore
-def test_live_s3_lifecycle(live_manager: storage.S3StorageManager, require_secrets: bool) -> None:
+@pytest.mark.parametrize("prefix", ["", "my/test/prefix/"])  # type: ignore
+def test_live_s3_lifecycle(
+    live_manager: storage.S3StorageManager, require_secrets: bool, prefix: str
+) -> None:
     def post_delete_cb(storage_id: str) -> None:
         """Search s3 directly to ensure that a checkpoint is actually deleted."""
         found = [obj.key for obj in live_manager.bucket.objects.filter(Prefix=storage_id)]
         if found:
             file_list = "    " + "\n    ".join(found)
             raise ValueError(f"found {len(found)} files in bucket after delete:\n{file_list}")
 
+    live_manager.prefix = prefix
```

Comment:
@rb-determined-ai ^ this is the line that I don't like. 

This is ungood because I am setting the member var `prefix` from outside of the obj. There is manipulation done to the prefix in the S3StorageManager ctor, which is why it is difficult to test `None` being passed in. Perhaps, I should make it `__prefix`, i.e., private and have a set/getter. My reasoning is that, while we are validating the format of the `prefix` when we parse it from the config, is it possible that someone (or somecode) could use the `S3StorageManager` from outside of the context of an experiment (such as in PyTest)? ... and so, we need some way of validating the format of the prefix inside this class as well.

I am having trouble with using a fixture for the `S3StorageManager`.IMHO, we should be parametrizing the `S3StorageManager`'s creation. Perhaps I am unaware of some PyTest functionality which could help us solve this?

Reply:
Sure.  Replace the fixture with a function... I don't think the fixture really adds anything of value here.